presto-sqlserver/src/main/java/com/facebook/presto/plugin/sqlserver/SqlServerClient.java

```diff
@@ -27,36 +30,52 @@
 import java.sql.SQLException;
 
 import static com.facebook.presto.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+import static java.lang.String.format;
 
 public class SqlServerClient
         extends BaseJdbcClient
 {
+    private static final Joiner DOT_JOINER = Joiner.on(".");
+
     @Inject
     public SqlServerClient(JdbcConnectorId connectorId, BaseJdbcConfig config)
     {
         super(connectorId, config, "\"", new DriverConnectionFactory(new SQLServerDriver(), config));
     }
 
     @Override
-    public void commitCreateTable(JdbcOutputTableHandle handle)
+    protected void renameTable(String catalogName, SchemaTableName oldTable, SchemaTableName newTable)
     {
-        StringBuilder sql = new StringBuilder()
-                .append("sp_rename ")
-                .append(singleQuote(handle.getCatalogName(), handle.getSchemaName(), handle.getTemporaryTableName()))
-                .append(", ")
-                .append(singleQuote(handle.getTableName()));
+        try (Connection connection = connectionFactory.openConnection()) {
+            String sql = format(
+                    "sp_rename %s, %s",
+                    singleQuote(catalogName, oldTable.getSchemaName(), oldTable.getTableName()),
+                    singleQuote(newTable.getTableName()));
+            execute(connection, sql);
+        }
+        catch (SQLException e) {
+            throw new PrestoException(JDBC_ERROR, e);
+        }
+    }
 
-        try (Connection connection = getConnection(handle)) {
-            execute(connection, sql.toString());
+    @Override
+    public void renameColumn(JdbcTableHandle handle, JdbcColumnHandle jdbcColumn, String target)
+    {
+        try (Connection connection = connectionFactory.openConnection()) {
+            String sql = format(
+                    "sp_rename %s, %s, 'COLUMN'",
+                    singleQuote(handle.getCatalogName(), handle.getSchemaName(), handle.getTableName(), jdbcColumn.getColumnName()),
+                    singleQuote(target));
+            execute(connection, sql);
         }
         catch (SQLException e) {
             throw new PrestoException(JDBC_ERROR, e);
         }
     }
 
-    private static String singleQuote(String catalog, String schema, String table)
+    private static String singleQuote(String... objects)
     {
-        return singleQuote(catalog + "." + schema + "." + table);
+        return singleQuote(DOT_JOINER.join(objects));
```

Comment:
I would inline `DOT_JOINER` for readability. I don't think there's much to gain by having it be a constant.